# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/PBackground.ipdl
# Commit: 7a4e3f5f674a
# Full Hash: 7a4e3f5f674ad675fa084bf1f30a79a15e8c7d2a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r 9da00c1364a5 -r 7a4e3f5f674a ipc/glue/PBackground.ipdl
--- a/ipc/glue/PBackground.ipdl	Wed Jan 18 00:50:19 2023 +0000
+++ b/ipc/glue/PBackground.ipdl	Wed Jan 18 00:50:19 2023 +0000
@@ -53,6 +53,7 @@
 include protocol PRemoteDecoderManager;
 include protocol PWebSocketConnection;
 include protocol PWebTransport;
+include protocol PFetch;
 
 include DOMTypes;
 include IPCBlob;
@@ -131,6 +132,7 @@
   manages PUDPSocket;
   manages PVerifySSLServerCert;
   manages PVsync;
+  manages PFetch;
 
 parent:
   // Only called at startup during mochitests to check the basic infrastructure.
@@ -322,6 +324,8 @@
 
   async PIPCClientCerts();
 
+  async PFetch();
+
 child:
   async PCache();
   async PCacheStreamControl();
